lib/api/objectCopy.js

```diff
@@ -318,14 +316,14 @@ function objectCopy(authInfo, request, sourceBucket,
             // also skip if 0 byte object, unless location constraint is an
             // external backend and differs from source, in which case put
             // metadata to backend
-            let locationType;
+            let sourceLocationConstraintType;
             if (storeMetadataParams.metaHeaders[locationHeader]) {
-                locationType = config.getLocationConstraintType(
+                sourceLocationConstraintType = config.getLocationConstraintType(
                   storeMetadataParams.metaHeaders[locationHeader]);
             }
             // NOTE: remove the following when we will support putting a
             // versioned object to a location-constraint of type AWS or Azure.
-            if (constants.externalBackends[locationType]) {
+            if (constants.externalBackends[sourceLocationConstraintType]) {
```

Comment:
@electrachong, note this will have to be updated.  We should move it to the wrapper when you change it to be just azure.